conjure-core/src/main/java/com/palantir/parsec/StringParserState.java

```diff
@@ -47,18 +44,9 @@ public int next() {
     }
 
     @Override
-    public void mark() {
-        marks.push(current);
-    }
-
-    @Override
-    public void rewind() {
-        current = marks.pop();
-    }
-
-    @Override
-    public void release() {
-        marks.pop();
+    public MarkedLocation mark() {
+        int markedPosition = current;
+        return () -> current = markedPosition;
```

Comment:
I like this refactoring, makes it easier to reason about the marked locations!

Reply:
Yep :-) I ran into an issue in my initial fix where a rewind call didn't rewind to the expected mark (resulting in an incorrect value, `map<string` was parsed as `string`), so I figured this refactor would be safer than trying to grok all the mark/rewind pairs!